### PR TITLE
Add some workflows

### PR DESCRIPTION
**Pull request by avik-pal**
Some useful workflows from ChainRules

### CODE REVIEW

File: .github/workflows/VersionVigilante.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@latest
+      - name: VersionVigilante.main
+        id: versionvigilante_main
+        run: |
+          using Pkg
+          Pkg.add("VersionVigilante")
+          using VersionVigilante
+          VersionVigilante.main("https://github.com/${{ github.repository }}", master_branch="main")
```
Comment (avik-pal) [inline .github/workflows/VersionVigilante.yml:17]:
```suggestion
          VersionVigilante.main("https://github.com/${{ github.repository }}", master_branch="master")
```

**Comment by avik-pal** [comment]
@ChrisRackauckas can you create a label for "needs version bump"? Or give me merge rights to the repo

File: .github/workflows/VersionVigilante.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: VersionVigilante
```
Comment (ChrisRackauckas) [inline .github/workflows/VersionVigilante.yml:1]: What's this?

Comment (avik-pal) [reply .github/workflows/VersionVigilante.yml:1]: screams if you:
1. forget to update the version
2. updated the version twice or more over the master

**Pull request closed without merging**